### DISCUSSION TITLE
Custom dropdown for select all rows checkbox

### DISCUSSION THREAD

**Discussion by sanchit3008** [TanStack/table#3737] [Q&A]
I need to build a feature such taht when user clicks on select all rows checkbox in datagrid, they should see a menu with 3 options - to deselect all rows currently selected, to select all rows on the visible page and to select all rows across all pages.

Is there any way I can achieve this behaviour upon checkbox click?

**Comment by alessandro308** [comment]
I think you shuld create a custom cell render that shows the dropdown and then invoke the functions to select or deselect according to the user selection.
